src/inputs/DateField.tsx

```diff
@@ -30,160 +29,146 @@ export interface DateFieldProps {
-          // Turn gray when disabled
-          ...(disabled ? Css.gray400.bgGray100.cursorNotAllowed.$ : {}),
-          ...(errorMsg ? Css.bRed600.$ : {}),
-        },
-        // Highlight on focus
-        "& .DayPickerInput input:focus-within": Css.bLightBlue700.$,
-        // The S / M / T / W ... heading
-        "& .DayPicker-Weekday": Css.pPx(8).xs.gray400.important.$,
-        // Un-collapse the borders so we can hover each cell
-        "& .DayPicker-Month": Css.add({ borderCollapse: "separate" }).$,
-        // Make the boxes smaller, this ends up being 32x32 which matches figma
-        "& .DayPicker-Day": Css.pPx(8).xs.ba.bWhite.br4.$,
-        // For today, use a background
-        "& .DayPicker-Day--today": Css.bgGray100.$,
-        // For selected, use a background
-        "& .DayPicker-Day--selected": Css.bgLightBlue700.white.$,
-        // For pressed
-        "& .DayPicker-Day:active": Css.bgGray400.$,
-        // Render over our calendar icon decoration in InputElement
-        "& .DayPickerInput-Overlay": Css.br4.z2.topPx(4).$,
-        // Make the month title, i.e. "May 2021", match figma; pyPx nudge matches the NavbarElement nudging
-        "& .DayPicker-Caption > div": Css.base.pyPx(2).$,
-      }}
-    >
-      {label && <Label labelProps={labelProps} label={label} suffix={labelSuffix} />}
-      <DayPickerInput
-        onDayChange={onChange as any}
-        dayPickerProps={{
-          navbarElement: NavbarElement,
-          weekdayElement: Weekday,
-        }}
-        component={InputElement}
-        // inputProps comes from react-aria and is how end up getting things like
-        // disabled=true passes through to our InputElement
-        inputProps={mergeProps(inputProps, { onFocus, onBlur })}
-        value={value}
-        placeholder=""
-        formatDate={formatDate}
-        parseDate={parseDate}
-        format={format}
-      />
-      {errorMsg && <ErrorMessage errorMsg={errorMsg} />}
-      {helperText && <HelperText helperText={helperText} />}
-    </div>
+  const inputWrapRef = useRef<HTMLDivElement | null>(null);
+  const buttonRef = useRef<HTMLButtonElement | null>(null);
+  const overlayRef = useRef<HTMLDivElement | null>(null);
+  const [inputValue, setValue] = useState(
+    value ? (long && readOnly ? dateFnsFormat(value, longFormat) : formatDate(value)) : "",
   );
-}
+  const textFieldProps = {
+    ...others,
+    label: label ?? "date",
+    isDisabled: disabled,
+    isReadOnly: false,
+    "aria-haspopup": "dialog" as const,
+    value: inputValue,
+  };
+  const state = useOverlayTriggerState({});
+  const { labelProps, inputProps } = useTextField(
+    {
+      ...textFieldProps,
+      // Open on focus of the input.
+      onFocus: (e) => {
+        state.toggle();
+        maybeCall(onFocus, e);
+      },
+      onBlur: (e) => {
+        // If we are interacting any other part of the inputWrap (such as the calendar button) return early
```

Comment:
Huh...so if it goes something like:

* focus input
* change focus to the calendar button
* this onBlur fires but early returns/does nothing b/c inputWrap.Ref.contains(related) returns true 
* change focus to some other field
* this onBlur isn't fired b/c it's hooked up directly to the `input` and not the `inputWrapper`

Will we miss the if-`!parseDate`-reset-the-date logic? How does that get called? Does it need to be?